Support for native simulator

flimberger opened the pull request:
Add support for the `native_sim` board. This will simplify testing changes to the UI and epub logic which don't relate to hardware.

flimberger committed:
feat: Separate configuration for hardware and native_sim

Split hardware-specific Kconfig options from the project configuration.
Introduce device-tree overlay and Kconfig options for native_sim.

Signed-off-by: Florian Limberger <flo@purplekraken.com>

flimberger committed:
fix: Prevent various crashes when there is no book

Signed-off-by: Florian Limberger <flo@purplekraken.com>

flimberger committed:
fix: Use correct format specifiers for size_t

This does not show up on 32-bit ARM, although the compiler should warn
about signedness anyway.

Signed-off-by: Florian Limberger <flo@purplekraken.com>

flimberger committed:
fix: Add missing include for snprintf

`Snprintf(3)` is declared in `<string.h>`, so include it.

Signed-off-by: Florian Limberger <flo@purplekraken.com>

flimberger committed:
fix: Silence bidirectional character warning

In C code for the font, bidirectional characters occur as part of the
character set, so silence the warning for the inclusion of the header
file.

Signed-off-by: Florian Limberger <flo@purplekraken.com>

Allegra42 commented on app/boards/native_sim_64.overlay:
```diff
@@ -0,0 +1,69 @@
+
+&sdl_dc {
+	compatible = "zephyr,sdl-dc";
+	height = <480>;
```
Undecided: the current orientation is a result of missing driver support for portrait for the used display. We could build the emulator in portrait mode to test and emulate the intended format until the driver is fixed

flimberger replied:
I used these values to emulate the real hardware more closely. I can look into rotating the display, but I think this would best be added as a separate feature.

Allegra42 commented on app/boards/native_sim_64.overlay:
```diff
@@ -0,0 +1,69 @@
+};
+
+/ {
+	keys: keys {
```
May we could introduce a configuration option for the hardware buttons as the code is intended to work with a touchscreen as well. This would avoid the need to add the button definitions for the emulator

flimberger committed:
fix: Rework documentation

Restructure README for clarity and a clear way how to get started with
the emulator. Add a dedicated section for the emulator to the project
documentation.

Signed-off-by: Florian Limberger <flo@purplekraken.com>

flimberger commented on README.md:
````diff
@@ -108,6 +108,36 @@ With Blackmagic Debug Probe using the SWD interface:
+```
+
+Alternatively, if the FUSE development headers are installed,
+FUSE access to the simulated storage can be enabled:
````
clarified the wording
